src/util/pelz_json_parser.c

```diff
@@ -21,19 +81,13 @@ int request_decoder(charbuf request, RequestType * request_type, charbuf * key_i
   memcpy(str, request.chars, request.len);
   json = cJSON_Parse(str);
   free(str);
-  if (!cJSON_HasObjectItem(json, "request_type"))
+  if(get_JSON_int_field(json, "request_type", (int*)request_type))
   {
     pelz_log(LOG_ERR, "Missing required JSON key: request_type.");
     cJSON_Delete(json);
     return (1);
   }
-  else if (!cJSON_IsNumber(cJSON_GetObjectItem(json, "request_type")))
-  {
-    pelz_log(LOG_ERR, "Incorrect data type of JSON value of JSON key: request_type. Data type should be integer.");
-    cJSON_Delete(json);
-    return (1);
-  }
-  *request_type = (RequestType) cJSON_GetObjectItemCaseSensitive(json, "request_type")->valueint;
+
   switch (*request_type)
```

Comment:
We can remove switch statement and just check and get ```key_id``` and ```data```

Reply:
Mostly fixed in 2e361dc, but that left a typo that got fixed in 344c145.